src/python/pants/backend/python/goals/package_pex_binary.py

```diff
@@ -76,22 +80,35 @@ async def package_pex_binary(
     field_set: PexBinaryFieldSet,
     pex_binary_defaults: PexBinaryDefaults,
     global_options: GlobalOptions,
+    dist_dir: DistDir,
 ) -> BuiltPackage:
-    resolved_entry_point = await Get(
+    entry_point = await Get(
         ResolvedPexEntryPoint, ResolvePexEntryPointRequest(field_set.entry_point, field_set.sources)
     )
     output_filename = field_set.output_path.value_or_default(
         field_set.address,
         file_ending="pex",
         use_legacy_format=global_options.options.pants_distdir_legacy_paths,
     )
+
+    if not entry_point.val:
+        logger.warning(
+            f"No entry point set for the `pex_binary` target {field_set.address}, so the binary "
+            "built via the `package` goal will create a PEX that acts similarly to a virtual "
+            f"environment's interpreter; for example, running "
+            f"`{dist_dir.relpath / output_filename}` will open a REPL (with access to all of the "
+            "binary's first and third-party dependencies).\n\nDid you mean to instead set the "
+            "`sources` field or `entry_point` field? See "
+            "https://www.pantsbuild.org/docs/python-package-goal#creating-a-pex-file-from-a-pex_binary-target"
+        )
```

Comment:
Unavoidable warnings aren't great (and this would be unavoidable if this was the behavior you actually wanted).

How about making either `entry_point` or `sources` required, and supporting adding a magic value to trigger the behavior described in this warning message? i.e. `entry_point = '<repl>'` or something.

Reply:
We could do `entry_point = '<none>'` maybe? I wish that we could force you to say in the BUILD file `entry_point=None`, but we have no idea if it would be `None` because explicitly set vs. just the default.

The reason for `<none>` over `<repl>` is that this can be used for more than a REPL. E.g. you may want to feed this PEX into another PEX you have with `--pex-path`. Or run a script with the PEX.

Reply:
Sure, sounds good!